evennia/utils/option.py

```diff
@@ -0,0 +1,62 @@
+from evennia.utils.utils import string_partial_matching
+
+
+class OptionHandler(object):
+    """
+    This is a generic Option handler meant for Typed Objects - anything that implements AttributeHandler.
+
+    It uses a dictionary to store-and-cache frequently used settings such as colors for borders or an
+    account's timezone.
+    """
+
+    def __init__(self, obj, options_dict=None, save_category=None):
+        if not options_dict:
+            options_dict = dict()
+        self.options_dict = options_dict
+        self.save_category = save_category
+        self.obj = obj
+        self.options = dict()
+
+    def __getitem__(self, item):
+        if item not in self.options_dict:
+            raise KeyError("Option not found!")
+        if item in self.options:
+            return self.options[item]
+        import evennia
+        option_def = self.options_dict[item]
+        save_data = self.obj.attributes.get(item, category=self.save_category)
+        if not save_data:
+            return evennia.VALID_HANDLER[option_def[1]](option_def[2])
```

Comment:
I see no reason for having this 'nested handler' in here. Let the option handler itself load the available validators and apply them here. Also - this is the place to catch validation errors, not inside each validator.

Reply:
What if other arbitrary code wants to use those Validators without needing an Account object to reference? They're useful for more than just Options. :)

Reply:
Not sure what you mean by this - the validators have no dependency on any Account object? A validator's job is straight forward - parse a value  and either raise `ValueError` or return the expected type and form for that validator.